apps/src/blockly/addons/cdoFieldDropdown.js

```diff
@@ -90,7 +90,10 @@ export default class CdoFieldDropdown extends GoogleBlockly.FieldDropdown {
     // See CDO implementation at https://github.com/code-dot-org/blockly/blob/main/core/ui/fields/field_dropdown.js#L305
     this.config = element.getAttribute('config');
     if (this.config) {
-      this.menuGenerator_ = this.getUpdatedOptionsFromConfig(this.config);
+      this.menuGenerator_ = getUpdatedOptionsFromConfig(
+        this.config,
+        this.menuGenerator_
```

Comment:
This is slightly out of the scope of this PR, but since you're moving stuff around I think it's probably the right time to ask the question...it feels a little weird to me that we're passing this as an argument to itself (when it's not recursive or anything)?

Wondering if we could do the initial prep of existing options outside of `getUpdatedOptionsFromConfig`:

```
const existingOptions = Array.isArray(menuGenerator)
    ? menuGenerator.reduce((optionsMap, curr) => {
        optionsMap[curr[1]] = curr[0];
        return optionsMap;
      }, {})
    : {};
```

Then update your function to be `getUpdatedOptionsFromConfig(existingOptions, config)`?

Reply:
Yeah - that change would make sense. Will do - thanks!